lib/iris/coords.py

```diff
@@ -1580,29 +1654,29 @@ def _new_points_requirements(self, points):
         if len(points) > 1 and not iris.util.monotonic(points, strict=True):
             raise ValueError('The points array must be strictly monotonic.')
 
-    @property
-    def points(self):
-        """The local points values as a read-only NumPy array."""
-        return self._points_dm.data
-
-    @points.setter
-    def points(self, points):
-        # We must realise points to check monotonicity.
+    def _points_setter(self, points):
+        # DimCoord always realises the points, to allow monotonicity checks.
         copy = is_lazy_data(points)
         points = as_concrete_data(points)
-        points = np.array(points, copy=copy, ndmin=1)
+        # Ensure it is an actual array, and also make our own distinct view
+        # so that we can make it read-only.
+        points = np.array(points, copy=copy)
 
-        # Set or update DataManager.
-        if self._points_dm is None:
-            self._points_dm = DataManager(points)
-        else:
-            self._points_dm.data = points
+        # Invoke the generic points setter.
+        super(DimCoord, self)._points_setter(points)
+
+        if self._points_dm is not None:
+            # Re-fetch the core array, as the super call may replace it.
+            points = self._points_dm.core_data()
+            # N.B. always a *real* array, as we realised 'points' at the start.
 
-        # Checks for 1d, numeric, monotonic
-        self._new_points_requirements(self._points_dm.data)
+            # Check validity requirements for dimension-coordinate points.
+            self._new_points_requirements(points)
 
-        # Make the array read-only.
-        self._points_dm.data.flags.writeable = False
+            # Make the array read-only.
+            points.flags.writeable = False
```

Comment:
This will update the underlying array (i.e. `self._points_dm.core_data()`) as well, right?